scripts/apps/authoring/multiversion/directives/MultiversionDropdownDirective.js

```diff
@@ -0,0 +1,109 @@
+class LinkFunction {
+    constructor(multiVersion, desks, archiveService, scope, elem) {
+        this.multiVersion = multiVersion;
+        this.desks = desks;
+        this.archiveService = archiveService;
+        this.scope = scope;
+        this.elem = elem;
+
+        this.init();
+    }
+
+    /**
+     * @ngdoc method
+     * @name sdMultiversionDropdown#init
+     * @private
+     * @description Initializes the directive with default values for the scope
+     * and with necessary watchers.
+     */
+    init() {
+        this.scope.$watchGroup(['item._id', 'item._latest_version'], () => {
+            this.fetchVersions();
+        });
+
+        this.scope.toggle = this.toggle.bind(this);
+        this.scope.isSelected = this.isSelected.bind(this);
+        this.scope.open = this.open.bind(this);
+    }
+
+    /**
+     * @ngdoc method
+     * @name sdMultiversionDropdown#fetchVersions
+     * @private
+     * @description fetches all the versions of currently opened article.
+     */
+    fetchVersions() {
+        this.desks.initialize()
+        .then(() => {
+            this.scope.desks = this.desks.desks;
+            this.scope.stages = this.desks.deskStages;
+            this.archiveService.getVersionHistory(this.scope.item, this.desks, 'versions').then((versions) => {
```

Comment:
The thing I personally liked most about promises is that it helped us avoid many callbacks with JavaScript, by chaining them. Even though functionally the code is the same, I think it would look nicer like this:
```js
this.desks.initialize()
    .then(() => {
        this.scope.desks = this.desks.desks;
        this.scope.stages = this.desks.deskStages;

        return this.archiveService.getVersionHistory(this.scope.item, this.desks, 'versions');
    })
    .then((versions) => {
        // ...
    });
```
What do you think?